### PR TITLE
feat(1password): add plugin for 1password

### PR DESCRIPTION
**Pull request by r-medina**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

- adds a [1password](https://1password.com/) plugin

### CODE REVIEW

**Comment by r-medina** [comment]
just realized that `pbcopy` will only work on os x. going to update when i have time later this week

**Comment by bartekpacia** [comment]
Any update @r-medina?

File: plugins/1password/1password.plugin.zsh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# opswd puts the password of the named service into the clipboard. If there's a
+# one time password, it will be copied into the clipboard after 5 seconds. The
+# clipboard is cleared after another 10 seconds.
+function opswd() {
```
Comment (ianchesal) [inline plugins/1password/1password.plugin.zsh:4]: Was about to write this plugin when I saw you had it here already. One change that would be good would be to add completion loading into the plugin. So, at the top of the file:

```
if type op; then
     eval "$(op completion zsh)"; compdef _op op
fi
```

**Comment by ianchesal** [comment]
I have left a comment for adding in `op` completion into the plugin. Otherwise this looks excellent and would love to see it get merged.

**Comment by ianchesal** [comment]
Actually, I think all your `if type...` calls need stderr redirect to stdout and stdout dumped to `/dev/null` otherwise you end up with lots messages like:

```
op is /usr/local/bin/op
```

in your terminal with you use this plugin, right?

File: plugins/1password/1password.plugin.zsh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    if [[ $# < 1 ]]; then
+	echo "Usage: opswd <service>"
+	return 1
+    fi
+
+    if ! type pbcopy; then
```
Comment (ianchesal) [inline plugins/1password/1password.plugin.zsh:10]: Should be:

```
if ! type pbcopy  >/dev/null 2>&1; then
```

File: plugins/1password/1password.plugin.zsh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	if ! type xclip; then
```
Comment (ianchesal) [inline plugins/1password/1password.plugin.zsh:11]: Should be:

```
if ! type xclip  >/dev/null 2>&1; then
```

**Review by 53jk1 — APPROVED** [review]

**Comment by bartekpacia** [comment]
I'd love to see it merged!

File: plugins/1password/1password.plugin.zsh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  (
+    sleep 5 && op get totp "$service" 2>/dev/null | clipcopy 2>/dev/null
+    sleep 10 && clipcopy < /dev/null 2>/dev/null &
+  ) &!
+}
+
+function _opswd() {
+  local -a services
+  services=("${(@f)$(op list items --categories Login 2>/dev/null | jq -r '.[].overview.title')}")
```
Comment (mcornella) [inline plugins/1password/1password.plugin.zsh:32]: The `jq` call could also be removed with
```sh
op list items --categories Login 2>/dev/null | op get item - --fields title 2>/dev/null
```

**Review by mcornella — CHANGES_REQUESTED** [review]
I made code style changes and added a completion function for `opswd`, but there are other possible improvements.

**Pull request merged**